source-academy/frontend
# Implement read-only option to playground sessions

### Description

Adds an option to invite other users to a session as viewers (read-only mode) or editors.

### Changes to implementation
- Users can join a session using one of two IDs: the session editing ID or the session viewing ID. These IDs are only 6 characters long, allowing for easier sharing.
- URL paths for requests sent to the ShareDB backend now contain only the 6-character ID instead of the ShareDB document ID used in previous implementations (see [this PR](https://github.com/source-academy/sharedb-ace-backend/pull/25)).
- Details of the current session such as the document ID and read-only status are obtained by GET requests, stored in Redux, and then passed to the ShareDB-Ace client (see [this PR](https://github.com/source-academy/sharedb-ace/pull/1)).

### UI changes
- Previously, clicking on the invite button immediately creates a session. Now it shows either a Create button, or the session IDs if they are currently in a session.
- Upon successfully joining a session, user is greeted with a popup message stating their level of access (as an editor or viewer).
- Added labels to text inputs.


### Type of change


### How to test

To create a new session, click on Invite > Create. Open SA Playground in another tab and join using one of the session IDs (Join > enter your ID), after which a popup message will appear. If joining as editor, the Ace editor still works as normal. If joining as a viewer, however, the editor will be set to read-only, meaning users cannot change the text as if it were a text field. But users can still move their cursor or block parts of the code in the editor.

### Checklist
